distributed/shuffle/_worker_plugin.py

```diff
@@ -57,7 +54,7 @@ class ShuffleClosedError(RuntimeError):
     pass
 
 
-class ShuffleRun(Generic[T_transfer_shard_id, T_partition_id, T_partition_type]):
+class ShuffleRun(Generic[T_partition_id, T_partition_type]):
```

Comment:
Both array rechunking and dataframe shuffling now send data grouped by partition ID, so we can simplify this.